arangod/RocksDBEngine/RocksDBCuckooIndexEstimator.cpp

```diff
@@ -707,32 +732,126 @@ void RocksDBCuckooIndexEstimator<Key>::initializeDefault() {
   }
 }
 
+template<class Key>
+void RocksDBCuckooIndexEstimator<Key>::increaseMemoryUsage(
+    uint64_t value) noexcept {
+  _memoryUsage += value;
+  if (ADB_LIKELY(_memoryUsageMetric != nullptr)) {
+    _memoryUsageMetric->fetch_add(value);
+  }
+}
+
+template<class Key>
+void RocksDBCuckooIndexEstimator<Key>::decreaseMemoryUsage(
+    uint64_t value) noexcept {
+  TRI_ASSERT(_memoryUsage >= value);
+  _memoryUsage -= value;
+  if (ADB_LIKELY(_memoryUsageMetric != nullptr)) {
+    _memoryUsageMetric->fetch_sub(value);
+  }
+}
+
 template<class Key>
 void RocksDBCuckooIndexEstimator<Key>::deriveSizesAndAlloc() {
   _sizeMask = _niceSize - 1;
   _sizeShift = static_cast<uint32_t>((64 - _logSize) / 2);
 
+  // bytes for slots
+  std::size_t bytesForSlots = _size * kSlotSize * kSlotsPerBucket;
+  // we assume it to be a multiple of 64, cache-line aligned
+  TRI_ASSERT(bytesForSlots % 64 == 0);
+
+  std::size_t bytesForCounters = _size * kCounterSize * kSlotsPerBucket;
+
   // give 64 bytes padding to enable 64-byte alignment
-  _slotAllocSize = _size * kSlotSize * kSlotsPerBucket + 64;
+  _allocSize = bytesForSlots + bytesForCounters + 64;
 
-  _slotBase = new char[_slotAllocSize];
+  _allocBase = new char[_allocSize];
 
   _base = reinterpret_cast<char*>(
-      (reinterpret_cast<uintptr_t>(_slotBase) + 63) &
+      (reinterpret_cast<uintptr_t>(_allocBase) + 63) &
       ~((uintptr_t)0x3fu));  // to actually implement the 64-byte alignment,
```

Comment:
Just FYI there is [std::aligned_alloc](https://en.cppreference.com/w/cpp/memory/c/aligned_alloc). If you want to use `new` you can use
```
auto* p = new(std::align_val_t(64)) char[_allocSize];
```

However, some people on the internet claim that invoking `new` in that way, did not work as expected, although the standard mandates this.

Also `_allocBase` could be stored in a `std::unique_ptr` to make sure the data is always free eventually.

Feel free to completely ignore this comment.

Reply:
I moved the allocation to use `std::make_unique<char[]>(allocSize);` instead of `new`.
If there are concerns about `std::aligned_alloc`, then I think won't make that change in this PR.